optimum/intel/openvino/modeling_open_clip.py

```diff
@@ -0,0 +1,680 @@
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+
+        if isinstance(model_id, Path):
+            model_id = model_id.as_posix()
+
+        from_transformers = kwargs.pop("from_transformers", None)
+        if from_transformers is not None:
+            logger.warning(
+                "The argument `from_transformers` is deprecated, and will be removed in optimum 2.0.  Use `export` instead"
+            )
+            export = from_transformers
+
+        if len(model_id.split("@")) == 2:
+            if revision is not None:
+                logger.warning(
+                    f"The argument `revision` was set to {revision} but will be ignored for {model_id.split('@')[1]}"
+                )
+            model_id, revision = model_id.split("@")
+
+        config_path = config if isinstance(config, (str, os.PathLike)) else model_id
+        config = cls._load_config(
+            config_path,
+            revision=revision,
+            cache_dir=cache_dir,
+            token=token,
+            force_download=force_download,
+            subfolder=subfolder,
+            trust_remote_code=trust_remote_code,
+            local_files_only=local_files_only,
+        )
+
+        from_pretrained_method = cls._from_transformers if export else cls._from_pretrained
+        return from_pretrained_method(
+            model_id=model_id,
+            config=config,
+            revision=revision,
+            cache_dir=cache_dir,
+            force_download=force_download,
+            token=token,
+            subfolder=subfolder,
+            local_files_only=local_files_only,
+            trust_remote_code=trust_remote_code,
+            task=task,
+            **kwargs,
+        )
+
+
+@add_start_docstrings(
+    """
+    OpenVINO Model for OpenCLIP Text model for tasks like zero-shot-image-classification.
+    """,
+    MODEL_START_DOCSTRING,
+)
+class OVModelOpenCLIPText(OVModelOpenCLIPBase):
+    xml_model_name = "openvino_model_text.xml"
+
+    def __init__(self, model=None, config=None, tokenize_cfg=None, **kwargs):
+        super().__init__(model, config, **kwargs)
+        self.tokenize_cfg = tokenize_cfg
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        load_in_8bit: Optional[bool] = None,
+        quantization_config: Union[OVWeightQuantizationConfig, Dict] = None,
```

Comment:
could we add a test for quantization as well  ?

Reply:
added case to test_ovmodel_load_with_compressed_weights and test_ovmodel_load_with_uncompressed_weights